### PR TITLE
mpl: Static analysis fixes

### PR DESCRIPTION
**Pull request by raffenet**
## Pull Request Description

A couple issues found when testing out Clang `scan-build`.

## Expected Impact

none

## Author Checklist
* [x] Reference appropriate issues (with "Fixes" or "See" as appropriate)
* [x] Remove xfail from the test suite when fixing a test
* [x] Commits are self-contained and do not do two things at once
* [x] Commit message is of the form: `module: short description` and follows [good practice](https://chris.beams.io/posts/git-commit/)
* [x] Passes whitespace checkers
* [x] Passes warning tests
* [ ] Passes all tests
* [x] Add comments such that someone without knowledge of the code could understand
* [x] Add Devel Docs in the `doc/` directory for any new code design

### CODE REVIEW

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Comment by raffenet** [comment]
I have a typo fix for the second commit message queued up once the tests pass.

File: src/mpl/src/shm/mpl_shm.c [hunk 1]
```diff
@@ -34,8 +34,14 @@ int MPL_shm_hnd_deserialize(MPL_shm_hnd_t hnd, const char *str_hnd, size_t str_h
     int rc = MPL_SHM_SUCCESS;
     MPLI_shm_hnd_reset_val(hnd);
     rc = MPLI_shm_ghnd_alloc(hnd, MPL_MEM_SHM);
+    if (rc != MPL_SHM_SUCCESS)
+        goto fn_exit;
```
Comment (hzhou) [inline src/mpl/src/shm/mpl_shm.c]: I personally prefer `return rc;` directly, but I am OK with this.

Comment (raffenet) [reply src/mpl/src/shm/mpl_shm.c]: I think I like the early return version better, too. I'll resubmit along with typo fixes.

**Comment by raffenet** [comment]
test:jenkins/ch3/tcp

**Pull request merged**